src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -551,222 +509,190 @@ MulticopterAttitudeControl::publish_actuator_controls()
 }
 
 void
-MulticopterAttitudeControl::run()
+MulticopterAttitudeControl::Run()
 {
-	_gyro_count = math::constrain(orb_group_count(ORB_ID(sensor_gyro)), 1, MAX_GYRO_COUNT);
-
-	for (unsigned s = 0; s < _gyro_count; s++) {
-		_sensor_gyro_sub[s] = orb_subscribe_multi(ORB_ID(sensor_gyro), s);
+	if (should_exit()) {
+		_vehicle_angular_velocity_sub.unregister_callback();
+		exit_and_cleanup();
+		return;
 	}
 
-	/* wakeup source: gyro data from sensor selected by the sensor app */
-	px4_pollfd_struct_t poll_fds = {};
-	poll_fds.events = POLLIN;
+	perf_begin(_loop_perf);
 
-	const hrt_abstime task_start = hrt_absolute_time();
-	hrt_abstime last_run = task_start;
-	float dt_accumulator = 0.f;
-	int loop_counter = 0;
+	/* run controller on gyro changes */
+	vehicle_angular_velocity_s angular_velocity;
 
-	bool reset_yaw_sp = true;
-	float attitude_dt = 0.f;
+	if (_vehicle_angular_velocity_sub.update(&angular_velocity)) {
+		const hrt_abstime now = hrt_absolute_time();
 
-	while (!should_exit()) {
```

Comment:
JFYI Setting ignore whitespace changes makes it a lot easier to read because of the indentation change.

Reply:
Thanks that's helpful.